Modules/Product/Http/Services/ProductService.php

```diff
@@ -103,12 +110,114 @@ public function fetch($request): array
-            "unpblshprdcts_count" => $unpublishedCount,
-            "allprdcts_count" => $productsCount
+            "pblshprdcts_count" => $publishedProductsCount,
+            "unpblshprdcts_count" => $unpublishedProductsCount,
+            "allprdcts_count" => $allProductsCount
         );
 
 
+        return ['res' => true, 'msg' => "", 'data' => $data];
+    }
+
+    /**
+     * Fetch All products By logged brand
+     *
+     * @return array
+     */
+
+    public function getProducts(): array
+    {
+        $userId = auth()->user()->id;
+        $data = [];
+        $products = Product::where('user_id', $userId)->where('status', 'publish')->get();
+        foreach ($products as $v) {
+            $productStock = $v->stock;
+            $productVariations = Product::find($v->id)->productVariation()->where('status', '1')->where('stock','>', '0')->get();
+            $productVariationsCount = $productVariations->count();
+            if ($productVariationsCount > 0) {
+                $variantMinPrice = Product::find($v->id)->productVariation()->min('price');
+                $price = $variantMinPrice . '+';
+                $variantStock = Product::find($v->id)->productVariation()->sum('stock');
+                $availability = $variantStock > 0 ? 'in stock' : 'out of stock';
+                $productStock += $variantStock;
+            } else {
+                $price = $v->usd_wholesale_price;
+                $availability = $v->stock > 0 ? 'in stock' : 'out of stock';
+            }
+            $variations=[];
+            if(!empty($productVariations)) {
+                foreach ($productVariations as $pv) {
+                    $image = !empty($pv->image) ? $pv->image : $v->featured_image;
+                    $sku = !empty($pv->sku) ? $pv->sku : $v->sku;
+                    $stock = !empty($pv->stock) ? $pv->stock : $v->stock;
+                    $variableArr = [];
+                    if (!empty($pv->value1)) {
+                        $variableArr[] = $pv->value1;
+                    }
+                    if (!empty($pv->value2)) {
+                        $variableArr[] = $pv->value2;
+                    }
+                    if (!empty($pv->value3)) {
+                        $variableArr[] = $pv->value3;
+                    }
+                    $variations[] = array(
+                        'id' => $v->id,
+                        'usd_wholesale_price' => $pv->price,
+                        'usd_retail_price' => $pv->retail_price,
+                        'variant_id' => $pv->id,
+                        'variant' => implode('/', $variableArr),
+                        'name' => $pv->name,
+                        'sku' => $sku,
+                        'featured_image' => $image,
+                        'stock' => $stock
+                    );
+                }
+            }
+            $prepacks  = [];
+            $productPrepackages = Product::find($v->id)->productPrepack()->where('active', '1')->get();
```

Comment:
get this relation in same query too like Product::('productPrepack','productVariation')....
You should make only 1 request to DB and get all the required relations with that then use that data wherever needed.